apt/method_test.go

```diff
@@ -290,3 +290,71 @@ func TestAptMethodRunFail(t *testing.T) {
 		}
 	}
 }
+
+// Test the case where the remote-end closes its pipe:
+func TestAptMethodRunReaderEOF(t *testing.T) {
```

Comment:
Apologies for delays. I checked out this code and ran the tests, and as a first sanity test, I removed your EOF handling change, and these tests still pass. Can you add a test that exercises and breaks the pipe in a way that should be handled, to confirm your code is correctly handling it?

Reply:
I will see what I can come up with, sure. That said, this particular change I'm not positive has such a test. The major difference here is breaking out of an infinite loop in the event that these pipe-related errors propagate. So my knee-jerk is that there isn't an analogous negative test, or at least not a good one that doesn't involve too my ugly mirror and/or doing something like spinning threads and waits, which will be inherently brittle.